### PR TITLE
Input checks

### PR DESCRIPTION
**Pull request by cgntnr**
Checks and error messages for profile preferences are done except for Passions and Orientation pages. I had a problem on controlling the number of checked Chip Buttons in a Chip Group.

### CODE REVIEW

**Comment by cgntnr** [comment]
On second thought, since I cleaned up the repo (for my part) and added most of the checks, we should better merge this in order to avoid further conflicts.

File: app/src/main/java/ch/epfl/sdp/blindly/profile/Profile4.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        setContentView(R.layout.set_profile_4)
+    }
+
+    fun startProfile4_2Or5(view: View) {
+        val radioGroup = findViewById<RadioGroup>(R.id.radioGroup)
+        val radioButtonMore = findViewById<RadioButton>(R.id.sex3_more)
+
+        when {
+            //No radio button is checked
+            radioGroup.checkedRadioButtonId == -1 -> {
```
Comment (cmsigrist) [inline app/src/main/java/ch/epfl/sdp/blindly/profile/Profile4.kt:24]: Maybe remove this magic number and replace with a const?

Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/profile/Profile4.kt:24]: Got it, can be replaced with NONE to make it clearer!

File: app/src/main/java/ch/epfl/sdp/blindly/profile/Profile2.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import ch.epfl.sdp.blindly.R
+
+
+private const val min_length = 2
+private const val max_length = 20
```
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/profile/Profile2.kt:12]: Might be a good idea to rename these constants using capital letters

File: app/src/main/java/ch/epfl/sdp/blindly/profile/Profile3.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun startProfile4(view: View) {
+        findViewById<TextView>(R.id.warning_p3).visibility = View.INVISIBLE
+        val datePicker: DatePicker = findViewById<View>(R.id.datePicker) as DatePicker
+        val day: Int = datePicker.dayOfMonth
+        val month: Int = datePicker.month + 1
+        val year: Int = datePicker.year
+        val age = getAge(year, month, day)
+        if (age < 18) {
```
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/profile/Profile3.kt:30]: Magic number here, consider creating a constant value

File: app/src/main/java/ch/epfl/sdp/blindly/profile/Profile5.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import android.view.View
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import ch.epfl.sdp.blindly.R
+import com.google.android.material.chip.ChipGroup
+
+
+class Profile5 : AppCompatActivity() {
+
+    private val limitSelection = 3
```
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/profile/Profile5.kt:14]: Might be a good idea to rename this constant using capital letters

File: app/src/main/java/ch/epfl/sdp/blindly/profile/Profile6.kt [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.set_profile_6)
+    }
+
+    fun startProfile7(view: View) {
+        val radioGroup = findViewById<RadioGroup>(R.id.radioGroup_p6)
+        findViewById<TextView>(R.id.warning_p6).visibility = View.INVISIBLE
+
+        //nothing is checked
+        if (radioGroup.checkedRadioButtonId == -1) {
```
Comment (Xelowak) [inline app/src/main/java/ch/epfl/sdp/blindly/profile/Profile6.kt:22]: Maybe create a constant for -1 to make it clearer that it takes this value when nothing is checked

**Review by Xelowak — CHANGES_REQUESTED** [review]

File: app/src/main/java/ch/epfl/sdp/blindly/profile/Profile2.kt (hunk 2 above)
Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/profile/Profile2.kt]: Corrected in the final commit.

File: app/src/main/java/ch/epfl/sdp/blindly/profile/Profile5.kt (hunk 4 above)
Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/profile/Profile5.kt]: Corrected in the final commit.

File: app/src/main/java/ch/epfl/sdp/blindly/profile/Profile6.kt (hunk 5 above)
Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/profile/Profile6.kt]: Corrected in the final commit.

File: app/src/main/java/ch/epfl/sdp/blindly/profile/Profile3.kt (hunk 3 above)
Comment (cgntnr) [reply app/src/main/java/ch/epfl/sdp/blindly/profile/Profile3.kt]: Corrected in the final commit.

**Review by Xelowak — APPROVED** [review]
Perfect ! :)

**Pull request merged**